src/string_util.cpp

```diff
@@ -279,7 +278,7 @@ namespace libtorrent
 			else
 			{
 				iface.port = std::atoi(port.c_str());
-				if (iface.port <= 0 || iface.port > 65535) iface.port = -1;
+				if (iface.port < 0 || iface.port > 65535) iface.port = -1;
```

Comment:
I separated the commits to easy the review (but it's trivial), it's that the PR queue is getting bigger